eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -372,5 +373,53 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
     assert(new String(warn2.data.toArray).startsWith("invalid announcement sig"))
   }
 
+  test("establish transient connection") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val probe = TestProbe()
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(remoteNodeId, None)), Nil)
+    probe.send(peerConnection, message)
+    probe watch peerConnection
+    probe.expectTerminated(peerConnection, max = Duration(1500, MILLISECONDS))
+  }
+
+  def sleep(millis: Int): Unit = {
+    val probe = TestProbe()
+    system.scheduler.scheduleOnce(Duration(millis, MILLISECONDS), probe.ref, ())(system.dispatcher)
```

Comment:
If you import `import scala.concurrent.duration.DurationInt` you will have access to syntaxic sugars for creating `Duration` values, for exemple `100 milliseconds`, I recommend using those and having `sleep` directly take a `FiniteDuration` as argument.

Reply:
Done